### DISCUSSION TITLE
How to use encode

### DISCUSSION THREAD

**Discussion by NonlinearNimesh** [eugeneware/jpeg-js#103] [Q&A]
Hi, I am trying to get JPEG/PNG image from a Dicom image using Dicom parser.

Initially, i am reading the Dicom file and converting it into raw pixel data
```
var dicom = require('dicom-parser');
var fs = require('fs');
var jpeg = require('jpeg-js');

var dicomFileAsBuffer = fs.readFileSync('/Users/nimesh/Downloads/Work/1_Mine/TBClassification3.7/TBClassifications/test/1.01.868.8289.50550.763722.6242389.2076076331839821530808901804551332385943463553544.dcm');

var dataSet = dicom.parseDicom(dicomFileAsBuffer);

var rawImageData = new Uint8Array(dataSet.byteArray.buffer,
                                dataSet.elements.x00280004.dataOffset);
console.log(rawImageData)
```
Next after generating raw pixel data, i am trying to save it into an image file and for that i am using writefileSync.
```
var width = 640,
  height = 320;
var frameData = new Buffer(width * height * 4);
var i = 0;
while (i < frameData.length) {
  frameData[i++] = 0xff; // red
  frameData[i++] = 0x00; // green
  frameData[i++] = 0x00; // blue
  frameData[i++] = 0xff; // alpha - ignored in JPEGs
}
var rawImageData = {
  data: frameData,
  width: width,
  height: height,
};
var jpegImageData = jpeg.encode(rawImageData);
console.log(jpegImageData);
/*
{ width: 320,
  height: 180,

fs.writeFileSync('image.jpg', jpegImageData.data);
```

And, the image file is also generated but it is not opening, seems like the generated image is not what i have used as input instead it is coming all red 

i am following this this ```https://stackoverflow.com/questions/49511099/getting-image-from-dicom-dcm-file-in-node-js``` link.

Can anyone help me on this

Thanks.

**Comment by patrickhulce** [comment]
Your example code does not use `jpeg-js` @NonlinearNimesh 

I don't know what format `dicom-parser` returns but if it is RGBA pixels then you need to `let data = jpeg.encode(rawImageData)` before you `writeFileSync`

**Comment by NonlinearNimesh** [comment]
HIi @patrickhulce sorry i have updated the code and the link which i am following for the objective, so if you can please loo at the question....

**Comment by patrickhulce** [comment]
> instead it is coming all red

This is expected behavior for your example code. You aren't actually using any of the image pixel data from the dicom file. You're explicitly filling a pixel array with all red pixels.

**Comment by patrickhulce** [comment]
I'm going to go ahead and convert this issue to a discussion since it appears you are mostly looking for basic usage assistance. Perhaps someone in the community will have the bandwidth to assist you further (or try Stack Overflow where this sort of request has more eager help 😃 )

(Sidenote: I don't know much about dicom format but I would be surprised if they use uncompressed image data. You might want to start there to understand more deeply *what* dicom-parser is returning to you and how to convert it to raw pixel data. See their docs...

> Note that actually displaying DICOM images is quite complex due to the variety of pixel formats and compression algorithms that DICOM supports. If you are interested in displaying images, please take a look at the cornerstone library and the cornerstoneWADOImageLoader which uses this library to extract the pixel data from DICOM files and display the images with cornerstone library. You can find the actual code that extracts grayscale pixel data using this library here.
